src/libs/actions/PolicyConnections.ts

```diff
@@ -0,0 +1,87 @@
+import Onyx from 'react-native-onyx';
+import type {OnyxUpdate} from 'react-native-onyx';
+import * as API from '@libs/API';
+import type UpdateConnectionConfigParams from '@libs/API/parameters/UpdatePolicyConnectionConfigurationParams';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import * as ErrorUtils from '@libs/ErrorUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ConnectionName} from '@src/types/onyx/Policy';
+
+function updatePolicyConnectionConfig({
+    policyID,
+    connectionName,
+    settingName,
+    settingValue,
+    originalSettingValue,
+}: {
+    policyID: string;
+    connectionName: ConnectionName;
+    settingName: string;
+    settingValue: unknown;
+    originalSettingValue: unknown;
+}) {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [settingName]: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE,
+                },
+                connections: {
+                    [connectionName]: {
+                        config: {
+                            [settingName]: settingValue,
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [settingName]: null,
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                pendingFields: {
+                    [settingName]: null,
+                },
+                errorFields: {
+                    [settingName]: ErrorUtils.getMicroSecondOnyxError('workspace.connection.error.genericUpdate'),
+                },
+                connections: {
+                    [connectionName]: {
+                        config: {
+                            [settingName]: originalSettingValue,
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const parameters: UpdateConnectionConfigParams = {
```

Comment:
If you apply the changes suggested in UpdatePolicyConnectionConfigurationParams, then this would become

```suggestion
    const parameters: UpdateConnectionConfigParams<T, S> = {
```

Reply:
https://github.com/Expensify/App/pull/38006/commits/fb2f4f7d4aa2a362bbfca38581e2c87410c985f0